rholang/src/test/scala/coop/rchain/rholang/interpreter/RuntimeSpec.scala

```diff
@@ -0,0 +1,380 @@
+package coop.rchain.rholang.interpreter
+import java.nio.file.Files
+
+import coop.rchain.models.Expr.ExprInstance.EVarBody
+import coop.rchain.models.Var.VarInstance.{BoundVar, FreeVar}
+import coop.rchain.models._
+import coop.rchain.rholang.interpreter.Interpreter.EvaluateResult
+import coop.rchain.rholang.interpreter.errors.ReduceError
+import monix.execution.Scheduler.Implicits.global
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.collection.immutable.BitSet
+import scala.concurrent.duration._
+
+class RuntimeSpec extends FlatSpec with Matchers {
+  val mapSize     = 10L * 1024L * 1024L
+  val tmpPrefix   = "rspace-store-"
+  val maxDuration = 5.seconds
+
+  val runtime = Runtime.create(Files.createTempDirectory(tmpPrefix), mapSize)
+
+  val nonReadableErr = ReduceError("Trying to read from non-readable channel.")
+
+  "rho:io:stdout" should "not get intercepted" in {
+    // new s(`rho:io:stdout`) in { for (x <- s) { Nil } }
+    val interceptNormalized = Par(
```

Comment:
I am very much against using the AST directly, unless it affects the tests potency in securing the change. From the conversation and the test itself I don't see why using AST here would be necessary.

You mention:
> I think the same about having an AST as the input to the tests. However for that it would be nice to have Interpreter.normalizeTerm to be public. Can I make it public?

There is `Interpreter.buildNormlizedTerm`, can we use that?

@goral09 mentions a reducer would be needed, but:
1) I'm not sure why, if the error is produced during normalizaton
2) Even if not, let's just use the reducer. You can have a look at `IntepreterSpec` if you need an e2e-ish approach.

Reply:
The error is thrown during _reduce_ phase so it makes sense to test that and not lexing, parsing, normalizing and only then the logic.

Reply:
The error is thrown during _reduce_ phase so it makes sense to test that and not lexing, parsing, normalizing and only then the logic.

Reply:
>  it makes sense to test that and not lexing, parsing, normalizing and only then the logic

It's not *testing* the lexing, parsing and normalizing. It's just using them in the test's setup. Any error in them is just a test setup error, not a test failure in the strict sense (not an assertion error). Notice the red vs yellow failure icon in IDEA, for example.

The test input data is hardly parseable to the bare eye, and comments are often outdated even at the time they're being commited.

I stand by my claim that we should `Interpreter.buildNormalizedTerm` in setup, and then run the reducer.